File: backend/src/dtos/auth_dto.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;


#[derive(Debug, Serialize, Deserialize, ToSchema)]
pub struct RegisterPayload {
    pub first_name: String,
    pub last_name: String,
    pub email: String,
    pub password: String,
}

#[derive(Debug, Serialize, Deserialize, ToSchema)]
pub struct LoginPayload {
    pub email: String,
    pub password: String,
}

#[derive(Debug, Serialize, Deserialize, ToSchema)]
pub struct UserProfile {
    pub id: i64,
    pub first_name: String,
    pub last_name: String,
    pub email: String,
}


#[derive(Debug, Serialize, Deserialize, ToSchema)]
pub struct AuthResponse {
    pub token: String,
    pub user: UserProfile,  
}

File: frontend/app/dtos/auth.dto.ts
import { z } from 'zod'

export const LoginPayloadSchema = z.object({
  email: z.string().email('Invalid email address'),
  password: z.string().min(1, 'Password is required')
})

export type LoginPayload = z.infer<typeof LoginPayloadSchema>

export const RegisterPayloadSchema = z.object({
  first_name: z.string().min(1, 'First name is required'),
  last_name: z.string().min(1, 'Last name is required'),
  email: z.string().email('Invalid email address'),
  password: z.string().min(6, 'Password must be at least 6 characters')
})

export type RegisterPayload = z.infer<typeof RegisterPayloadSchema>

export const UserProfileSchema = z.object({
  id: z.number(),
  first_name: z.string(),
  last_name: z.string(),
  email: z.string()
})

export type UserProfile = z.infer<typeof UserProfileSchema>

export const AuthResponseSchema = z.object({
  token: z.string(),
  user: UserProfileSchema
})

export type AuthResponse = z.infer<typeof AuthResponseSchema>
